src/main/java/spoon/support/visitor/equals/CloneHelper.java

```diff
@@ -78,7 +78,7 @@
 		if (elements == null || elements.isEmpty()) {
 			return EmptyClearableSet.instance();
 		}
-		Set<T> others = new HashSet<>(elements.size());
+		Set<T> others = new LinkedHashSet<>(elements.size());
```

Comment:
Also note this change. Using a plain `HashSet` here causes set order to be lost again during cloning.